spec/java/util/StringBuilder.main.lsl

```diff
@@ -0,0 +1,1022 @@
+    {
+        k = n - i;
+        newStr[k] = action CALL_METHOD(this.storage, "charAt", [i]);
+        newStr[i] = action CALL_METHOD(this.storage, "charAt", [k]);
+    }
+
+
+    fun *.toString (@target self: StringBuilder): String
+    {
+        action NOT_IMPLEMENTED("no final decision");
+    }
+
+
+    // within java.lang.AbstractStringBuilder
+    fun *.capacity (@target self: StringBuilder): int
+    {
+        // This is right realization ?
+        // original: return value.length >> coder;
+        // we will have another result, because we don't know "coder" value;
+        result = this.length;
+    }
+
+
+    // within java.lang.AbstractStringBuilder
+    fun *.ensureCapacity (@target self: StringBuilder, minimumCapacity: int): void
+    {
+
+    }
+
+
+    // within java.lang.AbstractStringBuilder
+    fun *.length (@target self: StringBuilder): int
+    {
+        result = this.length;
+    }
+
+
+    // within java.lang.AbstractStringBuilder
+    fun *.charAt (@target self: StringBuilder, index: int): char
+    {
+        _checkIndex(index);
+        result = action CALL_METHOD(this.storage, "charAt", [index]);
+    }
+
+
+    // within java.lang.AbstractStringBuilder
+    fun *.setLength (@target self: StringBuilder, newLength: int): void
+    {
+        if (newLength < 0)
+        {
+            //val message: String = "String index out of range: " + action OBJECT_TO_STRING(newLength);
+            action THROW_NEW("java.lang.StringIndexOutOfBoundsException", []);
+        }
+        var i: int = 0;
+        val newStr: array<char> = action ARRAY_NEW("char", newLength);
+        action LOOP_FOR(i, 0, newLength, +1, _setNewLength_loop(i, newStr));
+
+        this.storage = action OBJECT_TO_STRING(newStr);
+        this.length = newLength;
+    }
+
+
+    @Phantom proc _setNewLength_loop (i: int, newStr: array<char>): void
+    {
+        if (i < this.length)
+            newStr[i] = action CALL_METHOD(this.storage, "charAt", [i]);
+        // TODO: problem with: ''
+        //else
+            // That's right ? Did I understand correctly ?
+        //    newStr[i] = '0';
+    }
+
+
+    // within java.lang.AbstractStringBuilder
+    fun *.setCharAt (@target self: StringBuilder, index: int, ch: char): void
+    {
+        _checkIndex(index);
+        val newStr: array<char> = action ARRAY_NEW("char", this.length);
+        var i: int = 0;
+        action LOOP_FOR(i, 0, this.length, +1, _setNewLength_loop(i, index, newStr, ch));
+        this.storage = action OBJECT_TO_STRING(newStr);
+    }
+
+
+    @Phantom proc _setNewLength_loop (i: int, index: int, newStr: array<char>, ch: char): void
+    {
+        if(i != index)
+            newStr[i] = action CALL_METHOD(this.storage, "charAt", [i]);
+        else
+            newStr[i] = ch;
+    }
+
+
+    // within java.lang.AbstractStringBuilder
+    fun *.trimToSize (@target self: StringBuilder): void
+    {
+
```

Comment:
Empty function. Please add a comment with a clarification on why it is like that.

Reply:
Fixed

Reply:
Action `NOT_IMPLEMENTED` is not a correct one here. Consider `DO_NOTHING` instead.